### ISSUE TITLE
I can't save a template

### ISSUE THREAD

**Issue by RicarKS** [xylo/intellij-postfix-templates#211]
I can create a new template in the settings as shown in the picture,
![image](https://user-images.githubusercontent.com/49403030/152947892-b1e5bd83-1647-4e1e-a1c0-feee4f049c2e.png)
but when I finish editing this file and save (ctrl+s)
![image](https://user-images.githubusercontent.com/49403030/152948158-03853742-a424-4db4-b808-55eea6b6c26d.png)
![image](https://user-images.githubusercontent.com/49403030/152948481-1c7872d1-eee4-43cd-a39a-dcb60528dd30.png)
if I open the settings again, the file disappears,
![image](https://user-images.githubusercontent.com/49403030/152948355-4691094c-6664-4f07-9fa6-4a4cfdf2bd28.png)
and when I click apply or ok in the settings , this file will be deleted
![idea64_shfiz4ugw1](https://user-images.githubusercontent.com/49403030/152949414-0e49279a-6a33-4bbc-a9ca-c5291ec03f22.gif)

Here are the steps for all operations
![idea64_YScka4Xai5](https://user-images.githubusercontent.com/49403030/152950193-20cb0327-4e6a-4f96-9d75-248abadf1315.gif)
My native language is not English. I translated the above sentence with the help of Google Translate. If there is any inconsistency, please forgive me

**Comment by RicarKS** [comment]
My IntelliJ version is 2021.3 build #IU-213.5744.223

**Comment by xylo** [comment]
Thanks for the bug report and sorry for my late reply.
There is definitely something wrong with the plugin in your case.  When you go into the settings of Cutstom Postfix Templates you should never see an empty list of templates.  It should look like this:
![grafik](https://user-images.githubusercontent.com/577869/187777928-f2650035-10df-420c-be9b-d690d3b95d37.png)

It seems that in your case the plugin fails already in the initialization (loading of the templates).  Normally when the first project is loaded the plugin loads all the templates and tries to update the web template.  But since you do not even see the nodes "Java", "Scala", ... in the templates tree after opening the plugin settings, I guess that this procedure fails in your case.

Does the same error occur in IDEA 2022.2.1?

**Comment by wolray** [comment]
Yes, it occurs in IDEA 2022.2.1 (Community Edition) (Windows 10, x64).
Would it be caused by some kind of file permission reason?

**Comment by xylo** [comment]
Good question.  So the file user.postfixTemplates could be created by the plugin?  Then it's not a file permission problem.

So my first guess is that the plugin fails to download https://raw.githubusercontent.com/xylo/intellij-postfix-templates/master/templates/webTemplateFiles.yaml or some of the web template files.

I guess I have to add some try catch blocks and some error messages to identify the cause of the problem.

**Comment by wolray** [comment]
The file was created by the plugin, and deleted by it after re-open the settings panel (which is empty like fresh new).
It might be something wrong like you mentioned, and some deleting process in a catch/finally block worked, I guess.

**Comment by xylo** [comment]
I managed to reproduce the bug today by disallowing IDEA to connect to the internet.  I was in a network environment that required a proxy configuration but I disabled this config in IDEA (I choose no proxy).  So any connection failed.  This is how I got the same result as in your pictures above.

So I guess you IDEA version does not have an internet connection.  Is this right?  Either this or IDEA's requests to https://row.githubusercontent.com/... is blocked.

**Comment by xlj44400** [comment]
I just update 2021.2.2 -> 2022.2.2 , had the same problem
